pkg/archive/changes.go

```diff
@@ -386,7 +386,21 @@ func ExportChanges(dir string, changes []Change) (Archive, error) {
 		// during e.g. a diff operation the container can continue
 		// mutating the filesystem and we can see transient errors
 		// from this
+	changesLoop:
 		for _, change := range changes {
+			for _, exclude := range excludes {
+				exclude := filepath.Join("/", exclude)
```

Comment:
Hello @duglin 
`ExportChanges` is called in `NaiveDiffDriver`, which is used when the storage-driver is set to vfs and overlay. If the storage-driver is set to aufs, `archive.TarWithOptions` will be called.
`archive.TarWithOptions` uses relative path to compare the exclude patterns, while `ExportChanges` uses absolute path.

The first if-sttm below is used to match `--exclude=/path/file/*` or `--exclude=/path/file*.txt`, but it can't match `--exclude=/directory/to/exclude`.  The `else` block tries to match the `--exclude=/director/to/exclude` pattern.